spec/index.md

```diff
@@ -5679,6 +5795,7 @@ We can model the execution of WebAssembly functions as stateful functions that h
       sysenv : Env;
       cycles_refunded : Nat;
       method_name : NoText | Text;
+      deadline : NoDeadline | Timestamp;
```

Comment:
This field is not initialized - I'd rather put it into sysenv : Env and initialize it in the abstract state machine above whenever Env is created.

Reply:
Intuitively, I'd expect this to be passed the same way that `arg` is, because it's defined by the caller (or by the system in case of ingress messages). But maybe I'm missing something?

Reply:
> Intuitively, I'd expect this to be passed the same way that arg is, because it's defined by the caller (or by the system in case of ingress messages).

Sure, that's an option, too, but then the types of canister methods must be adjusted by one additional argument. If the deadline is put into Env, no changes to canister method types are necessary.